src/gc/impl/conservative/gc.d

```diff
@@ -2467,10 +2473,23 @@ MARK:
                 start = stop;
             }
 
-            if (block)
```

Comment:
Git forced me to stage the comment and the if condition (on the comment below) as replacement for just this line.
The error was mine that I wrote the comments and changed the line thinking that I could commit it line by line.
I should have changed the if condition, pushed, then added the comments.

Reply:
You can always edit commits as much as you want/need. Just use `git rebase -i`, use `edit` on the commit you want to edit (for example the one where code is in the commit adding documentation to remove the changed code), then go to the commit where the code should be and add it back (if you are not forced to do so due to a conflict).

In any case, if you just prefer to squash everything together to avoid spending too much time on this I think it's OK too.